servant-client/test/Servant/ClientTestUtils.hs

```diff
@@ -145,6 +147,46 @@ instance ToDeepQuery Filter where
     , (["name"], Just (Text.pack name'))
     ]
 
+-----------------------------
+-- MultiVerb test endpoint --
+-----------------------------
+
+-- This is the list of all possible responses
+type MultipleChoicesIntResponses = 
+  '[ RespondEmpty 400 "Negative"
+   , Respond 200 "Even number" Bool
+   , Respond 200 "Odd number" Int
+   ]
+
+data MultipleChoicesIntResult
+  = NegativeNumber
+  | Even Bool
+  | Odd Int
+  deriving stock (Generic)
+  deriving (AsUnion MultipleChoicesIntResponses)
+    via GenericAsUnion MultipleChoicesIntResponses MultipleChoicesIntResult
+
+instance GSOP.Generic MultipleChoicesIntResult
+
+-- instance MultipleChoicesIntResponses ~ res => AsUnion res MultipleChoicesIntResult where
```

Comment:
```suggestion
-- instance AsUnion MultipleChoicesIntResponses MultipleChoicesIntResult where
```

as long as you don't mention `res` again in the instance body, the other thing is really just noise.